SYCL/ESIMD/api/functional/ctors/ctor_fill.hpp

```diff
@@ -0,0 +1,226 @@
+
+  template <typename DataT, int NumElems>
+  static void call_simd_ctor(DataT init_value, DataT step, DataT *out) {
+    esimd::simd<DataT, NumElems> simd_by_rval;
+    simd_by_rval = esimd::simd<DataT, NumElems>(init_value, step);
+    simd_by_rval.copy_to(out);
+  }
+};
+
+// Descriptor class for the case of calling constructor in const reference
+// context.
+class const_ref {
+public:
+  static std::string get_description() { return "const reference"; }
+
+  template <typename DataT, int NumElems>
+  static void call_simd_ctor(DataT init_value, DataT step, DataT *out) {
+    return call_simd_by_const_ref<DataT, NumElems>(
+        esimd::simd<DataT, NumElems>(init_value, step), out);
+  }
+
+private:
+  template <typename DataT, int NumElems>
+  static void
+  call_simd_by_const_ref(const esimd::simd<DataT, NumElems> &simd_by_const_ref,
+                         DataT *out) {
+    simd_by_const_ref.copy_to(out);
+  }
+};
+
+// Enumeration of possible values for base value and step that will be provided
+// into simd constructor.
+enum class init_val {
+  min,
+  max,
+  zero,
+  min_half,
+  max_half,
+  neg_inf,
+  nan,
+  positive,
+  negative
+};
+
+// Dummy kernel for submitting some code into device side.
+template <typename DataT, int NumElems, typename T, init_val BaseVal,
+          init_val StepVal>
+struct kernel_for_fill;
+
+// Constructing a value for step and base values that depends on input
+// parameters.
+template <typename DataT, init_val BaseVal> DataT get_base_value() {
+  if constexpr (BaseVal == init_val::min) {
+    return value<DataT>::lowest();
+  } else if constexpr (BaseVal == init_val::max) {
+    return value<DataT>::max();
+  } else if constexpr (BaseVal == init_val::zero) {
+    return 0;
+  } else if constexpr (BaseVal == init_val::positive) {
+    return static_cast<DataT>(1.25);
+  } else if constexpr (BaseVal == init_val::negative) {
+    return static_cast<DataT>(-10.75);
+  } else if constexpr (BaseVal == init_val::min_half) {
+    return value<DataT>::lowest() / 2;
+  } else if constexpr (BaseVal == init_val::max_half) {
+    return value<DataT>::max() / 2;
+  } else if constexpr (BaseVal == init_val::neg_inf) {
+    return -value<DataT>::inf();
+  } else if constexpr (BaseVal == init_val::nan) {
+    return value<DataT>::nan();
+  } else {
+    static_assert(BaseVal != BaseVal, "Unexpected base value value");
+  }
+}
+
+// Struct that will be used for invocating of simd constructor.
+template <init_val BaseVal, init_val Step> struct call_simd_struct {
+  template <typename DataT, int NumElems, typename TestCaseT>
+  esimd_tests::shared_vector<DataT>
+  call_simd(sycl::queue &queue, const std::vector<DataT> &ref_data) {
+    esimd_tests::shared_vector<DataT> result{NumElems,
+                                             shared_allocator<DataT>{queue}};
+    esimd_tests::shared_vector<DataT> shared_ref_data{
+        ref_data.begin(), ref_data.end(), shared_allocator<DataT>{queue}};
```

Comment:
```suggestion
  auto
  call_simd(sycl::queue &queue, const std::vector<DataT> &ref_data) {
    using vector_t = esimd_tests::shared_vector<DataT>;
    vector_t result{NumElems, shared_allocator<DataT>{queue}};
    vector_t shared_ref_data{
        ref_data.begin(), ref_data.end(), shared_allocator<DataT>{queue}};
```
Minor - you might avoid repeating the same type

Reply:
not actual commit, now only one shared vector using